components/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/builders/ProviderConfigBuilder.java

```diff
@@ -118,16 +118,33 @@ public OIDProviderConfigResponse buildOIDProviderConfig(OIDProviderRequest reque
         providerConfig.setEndSessionEndpoint(buildServiceUrl(IdentityConstants.OAuth.LOGOUT,
                 IdentityUtil.getProperty(IdentityConstants.OAuth.OIDC_LOGOUT_EP_URL)));
 
-        try {
-            providerConfig.setUserinfoSigningAlgValuesSupported(new String[] {
-                    OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(
-                            OAuthServerConfiguration.getInstance().getUserInfoJWTSignatureAlgorithm()).getName()
-            });
-        } catch (IdentityOAuth2Exception e) {
-            throw new ServerConfigurationException("Unsupported signature algorithm configured.", e);
+        List<String> supportedUserinfoSigningAlgValues = OAuthServerConfiguration.getInstance()
+                .getUserInfoJWTSignatureAlgorithms();
+        String userInfoJWTSignatureAlgorithm =  OAuthServerConfiguration.getInstance()
+                .getUserInfoJWTSignatureAlgorithm();
+        if (!supportedUserinfoSigningAlgValues.contains(userInfoJWTSignatureAlgorithm)) {
+            supportedUserinfoSigningAlgValues.add(userInfoJWTSignatureAlgorithm);
+        }
+        if (!supportedUserinfoSigningAlgValues.isEmpty()) {
```

Comment:
Can't we do this one time and keep the structure without doing it again and again

Reply:
Modifications were moved to the OAuthServerConfiguration class in afee548fc78bfd5671aa11bc90274caee366b930